### DISCUSSION TITLE
I want to know what package a java class Node be in.

### DISCUSSION THREAD

**Discussion by XenoAmess** [pmd/pmd#3907] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
I want to know what package a java class Node be in.

**Describe the solution you'd like**
I want to know what package a java class Node be in.

**Describe alternatives you've considered**

**Additional context**
I think it reasonable, because we can always get this information from the `package` grammar in java source.

**Comment by oowekyala** [comment, marked as answer]
`ASTCompilationUnit` has a method [getPackageName](https://docs.pmd-code.org/apidocs/pmd-java/6.43.0/net/sourceforge/pmd/lang/java/ast/ASTCompilationUnit.html#getPackageName()) which you can also use in XPath with the attribute `@PackageName`. To find the node in Java you have to get the root of the tree, by asking `getParent` until you find it. We have added a method `getRoot` to every Node in PMD 7 which we could backport to make that easier.

If that does not answer your question, please add more details about how and from where you want to get that information

**Comment by XenoAmess** [comment]
> `ASTCompilationUnit` has a method [getPackageName](https://docs.pmd-code.org/apidocs/pmd-java/6.43.0/net/sourceforge/pmd/lang/java/ast/ASTCompilationUnit.html#getPackageName()) which you can also use in XPath with the attribute `@PackageName`. To find the node in Java you have to get the root of the tree, by asking `getParent` until you find it. We have added a method `getRoot` to every Node in PMD 7 which we could backport to make that easier.
> 
> If that does not answer your question, please add more details about how and from where you want to get that information

@oowekyala got it! Thanks!
